action.yml

```diff
@@ -94,20 +73,78 @@ runs:
 
     - name: Get baseline project ID
       id: get_baseline_project_id
-      uses: fjogeleit/http-request-action@v1.16.1
+      uses: fjogeleit/http-request-action@v1.16.2
       with:
         url: 'https://api.snyk.io/rest/orgs/${{ fromJson(steps.get_snyk_org_id.outputs.response).data[0].id }}/projects?version=${{ inputs.snyk-api-version }}&target_reference=${{ github.base_ref }}&names=${{ env.snyk_project_name }}%3A${{ env.snyk_target_file }}'
         method: 'GET'
         customHeaders: '{"Authorization": "token ${{ inputs.snyk-token }}"}'
         retry: 3
 
+    - name: Choose baseline project ID
+      id: choose_baseline_project_id
+      shell: bash
+      run: |
+        retrieved_baseline_project_id=${{ fromJson(steps.get_baseline_project_id.outputs.response).data[0].id }}
+        echo "snyk_baseline_project_id=${retrieved_baseline_project_id:-'${{ inputs.snyk-baseline-project-id }}'}" >> "$GITHUB_OUTPUT"
+
+    - name: Get ignored issues
+      id: get_ignored_issues
+      uses: fjogeleit/http-request-action@v1.16.2
+      with:
+        url: 'https://api.snyk.io/rest/orgs/${{ fromJson(steps.get_snyk_org_id.outputs.response).data[0].id }}/issues?version=${{ inputs.snyk-api-version }}&limit=100&ignored=true&scan_item.id=${{ steps.choose_baseline_project_id.outputs.snyk_baseline_project_id }}&scan_item.type=project'
+        method: 'GET'
+        customHeaders: '{"Authorization": "token ${{ inputs.snyk-token }}"}'
+        responseFile: 'ignored_issues_${{ env.snyk_project_name }}.json'
+        retry: 3
+
+    - name: Create ignore file based on already ignored issues
+      id: create_ignore_file
+      shell: bash
+      run: |
+        snyk auth ${{ inputs.snyk-token }}
+        mkdir ${{ env.snyk_project_name }}-policy
+        jq '.data[] | .attributes.key' ignored_issues_${{ env.snyk_project_name }}.json | while read issue_id; do
+          snyk ignore --id=$issue_id --policy-path=${{ env.snyk_project_name }}-policy
+        done
+
+    - name: Run Snyk with the snyk ignore file
```

Comment:
We now run synk test twice, is the first run really necessary?

Reply:
It is, for now. I plan to revise the input arguments for this action in a major release later on. This is to avoid changing all the snyk workflows right now. This is a tradeoff that should be ok.